product_supplierinfo_import/wizards/product_supplierinfo_import.py

```diff
@@ -0,0 +1,288 @@
+        comodel_name="res.partner",
+        string="Supplier",
+        compute="_compute_supplier_id",
+        store=True,
+        readonly=False,
+    )
+    date_start = fields.Date(string="Validity", required=True)
+    delay = fields.Integer()
+    create_new_products = fields.Boolean(
+        help="If a product isn't found by its Search Field, it will be created with "
+        "the provided data",
+        default=True,
+    )
+    supplierinfo_file = fields.Binary(required=True)
+    supplierinfo_filename = fields.Char()
+    template_id = fields.Many2one(
+        comodel_name="product.supplierinfo.import.template",
+        compute="_compute_template_id",
+        store=True,
+    )
+
+    @api.depends("template_id")
+    def _compute_supplier_id(self):
+        for record in self:
+            if record.template_id.supplier_id:
+                record.supplier_id = record.template_id.supplier_id
+
+    @api.depends("supplierinfo_file")
+    def _compute_template_id(self):
+        for record in self:
+            if record.supplierinfo_file:
+                filename = record.supplierinfo_filename or ""
+                if parse_version(xlrd.__version__) >= parse_version(
+                    "2.0"
+                ) or not filename.lower().endswith(".xlsx"):
+                    raise UserError(self.env._("Only .xlsx files are supported."))
+                data = base64.b64decode(record.supplierinfo_file)
+                workbook = xlrd.open_workbook(file_contents=data)
+                record._detect_template(workbook)
+            else:
+                record.template_id = False
+
+    @api.model
+    def _parse_header(self, header):
+        # Trim left and right blank chars and convert newlines for better matching
+        return [str(h).replace("\n", " ").strip() for h in header]
+
+    def _detect_template(self, workbook):
+        """Detect the template to be used from the sheet header"""
+        templates = self.env["product.supplierinfo.import.template"].search([])
+        template_headers = [(t, t._template_headers()) for t in templates]
+        header_values = []
+        for template, header in template_headers:
+            if template.sheet_number - 1 >= workbook.nsheets:
+                raise UserError(
+                    self.env._(
+                        f"Sheet number {template.sheet_number} is out of range. "
+                        f"The workbook only has {workbook.nsheets} sheets."
+                    )
+                )
+            sheet = workbook.sheet_by_index(template.sheet_number - 1)
+            header_values = self._parse_header(sheet.row_values(template.header_offset))
+            if set(header_values) == header:
+                self.template_id = template
+                return header_values
+        if not self.template_id:
+            raise UserError(
+                self.env._(
+                    f"No matching template for these header columns.\n"
```

Comment:
No format string is necessary in this line.
```suggestion
                    "No matching template for these header columns.\n"
```